beacon-chain/rpc/beaconv1/validator_test.go

```diff
@@ -0,0 +1,316 @@
+		ids := [][]byte{pubkey1[:], []byte("90"), pubkey3[:], []byte("129")}
+		resp, err := s.ListValidatorBalances(ctx, &ethpb.ValidatorBalancesRequest{
+			StateId: []byte("head"),
+			Id:      ids,
+		})
+		require.NoError(t, err)
+		for i, val := range resp.Data {
+			assert.Equal(t, idNums[i], val.Index)
+			assert.Equal(t, params.BeaconConfig().MaxEffectiveBalance, val.Balance)
+		}
+	})
+}
+
+func TestListCommittees(t *testing.T) {
+	ctx := context.Background()
+
+	var state iface.BeaconState
+	state, _ = testutil.DeterministicGenesisState(t, 8192)
+
+	t.Run("Head All Committees", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		resp, err := s.ListCommittees(ctx, &ethpb.StateCommitteesRequest{
+			StateId: []byte("head"),
+		})
+		require.NoError(t, err)
+		assert.Equal(t, int(params.BeaconConfig().SlotsPerEpoch)*2, len(resp.Data))
+		for _, datum := range resp.Data {
+			assert.Equal(t, true, datum.Index == types.CommitteeIndex(0) || datum.Index == types.CommitteeIndex(1))
+		}
+	})
+
+	t.Run("Head All Committees of Slot 4", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		slot := types.Slot(4)
+		resp, err := s.ListCommittees(ctx, &ethpb.StateCommitteesRequest{
+			StateId: []byte("head"),
+			Slot:    &slot,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, 2, len(resp.Data))
+		index := types.CommitteeIndex(0)
+		for _, datum := range resp.Data {
+			assert.Equal(t, slot, datum.Slot)
+			assert.Equal(t, index, datum.Index)
+			index++
+		}
+	})
+
+	t.Run("Head All Committees of Index 1", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		index := types.CommitteeIndex(1)
+		resp, err := s.ListCommittees(ctx, &ethpb.StateCommitteesRequest{
+			StateId: []byte("head"),
+			Index:   &index,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, int(params.BeaconConfig().SlotsPerEpoch), len(resp.Data))
+		slot := types.Slot(0)
+		for _, datum := range resp.Data {
+			assert.Equal(t, slot, datum.Slot)
+			assert.Equal(t, index, datum.Index)
+			slot++
+		}
+	})
+
+	t.Run("Head All Committees of Slot 2, Index 1", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		index := types.CommitteeIndex(1)
+		slot := types.Slot(2)
+		resp, err := s.ListCommittees(ctx, &ethpb.StateCommitteesRequest{
+			StateId: []byte("head"),
+			Slot:    &slot,
+			Index:   &index,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(resp.Data))
+		for _, datum := range resp.Data {
+			assert.Equal(t, slot, datum.Slot)
+			assert.Equal(t, index, datum.Index)
+		}
+	})
+
+	t.Run("Hex root not found", func(t *testing.T) {
```

Comment:
Can you copy/paste this test into other functions?

Reply:
Done!